app/models/devolved_administration_availability.rb

```diff
@@ -4,4 +4,27 @@ class DevolvedAdministrationAvailability
   embedded_in :local_transaction_edition
   field :type, type: String, default: "local_authority_service"
   field :alternative_url, type: String
+
+  validates :type, inclusion: { in: %w[local_authority_service devolved_administration_service unavailable], allow_blank: true }
+  validate :alternative_url_presence
+  validate :alternative_url_format
```

Comment:
With alternative_url_format we should have a think about what should happen to an alternative_url when someone changes the type. 

For example in the current situation I could set an alternative_url to be an erroneous URL but by changing the type we bypass the validation and have it be persisted. Demo:

```
irb(main):002:0> availability = DevolvedAdministrationAvailability.new(type: "devolved_administration_service", alternative_url: "/wrong-url")
=> #<DevolvedAdministrationAvailability _id: 60b7af5a39fad80caa4ef0a5, type: "devolved_administration_service", alternative_url: "/wrong-url">
irb(main):003:0> availability.valid?
=> false
irb(main):004:0> availability.type = "unavailable"
=> "unavailable"
irb(main):005:0> availability.valid?
=> true
irb(main):006:0> availability.alternative_url
=> "/wrong-url"
```

I'd imagine the simplest approach would be to always validate the alternative_url if it is present.

Reply:
Thanks, I updated to always validate the alternative_url if it is present.